pkgs/by-name/pr/protonmail-desktop/package.nix

```diff
@@ -9,28 +9,29 @@
 }:
 
 let
+  darwinArch = if stdenv.hostPlatform.system == "x86_64-darwin" then "x64" else "arm64";
   mainProgram = "proton-mail";
+  universalDarwinHash = "sha256-JfZwHFp0aZtHcbP7tyG7uqFs2w+LWKfnfyuxxpxDJZ8=";
+
 in
 stdenv.mkDerivation rec {
   pname = "protonmail-desktop";
-  version = "1.0.4";
+  version = "1.0.5";
 
-  src =
-    {
-      "x86_64-linux" = fetchurl {
-        url = "https://github.com/ProtonMail/inbox-desktop/releases/download/v${version}/proton-mail_${version}_amd64.deb";
-        hash = "sha256-KY/rjiJozOQW27FYljy5N1VKuKroJz3V485DPaH01JY=";
-      };
-      "x86_64-darwin" = fetchurl {
-        url = "https://github.com/ProtonMail/inbox-desktop/releases/download/v${version}/Proton.Mail-darwin-x64-${version}.zip";
-        hash = "sha256-I5Yj1JR3DaAmC6WKI4X/d/q9rvmsck9SE3Mx3AY6yvU=";
-      };
-      "aarch64-darwin" = fetchurl {
-        url = "https://github.com/ProtonMail/inbox-desktop/releases/download/v${version}/Proton.Mail-darwin-arm64-${version}.zip";
-        hash = "sha256-j1F8hhLSq/C1WQXGrYnvFK8nNz4qwoA1ohNzPsS3tiY=";
-      };
-    }
-    .${stdenv.hostPlatform.system};
+  src = fetchurl {
+    url =
+      if stdenv.isDarwin then
+        "https://github.com/ProtonMail/inbox-desktop/releases/download/${version}/Proton.Mail-darwin-${darwinArch}-${version}.zip"
+      else
+        "https://github.com/ProtonMail/inbox-desktop/releases/download/${version}/proton-mail_${version}_amd64.deb";
+    sha256 =
+      {
+        x86_64-linux = "sha256-En5vkTHYtwN6GMgbtyhzsPqknOPRO9KlTqZfbBFaIFQ=";
```

Comment:
How about moving this to a let binding as well? I'm not sure how this fares in nix world but this kind of incoherence is making me uneasy 🙂 other than this small nitpick (which can be ignored) this LGTM